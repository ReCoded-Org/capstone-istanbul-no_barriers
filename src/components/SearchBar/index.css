:root {
  --main-searchBar-bg-color: #eff4f6;
  --searchBar-text-color: #002736;
  --orange-link: #f2c94c;
  --dark-orange: #d88315;
  --orange: #fc9e01;
  --inputBar-background: #f4fcff;
}

.globalSearch {
  background-color: var(--main-searchBar-bg-color);
  color: var(--searchBar-text-color);
  padding: 5rem 6rem 4rem;
}

.searchBar {
  padding-top: 2rem;
}

.searchBarInput {
  padding: 0 1rem;
  margin-right: 1rem;
  border-color: var(--orange);
  border-radius: 0.3rem;
  background: var(--inputBar-background);
}

.searchBarInput:focus {
  border-style: inset;
  outline-color: var(--orange);
}

.searchBarInput::placeholder {
  color: var(--orange-link);
}

.searchBarButton {
  padding: 0.5rem 3rem;
  border-radius: 5rem;
}

.searchBarButton,
.searchBarButton:hover,
.searchBarButton:focus {
  background-color: var(--orange);
  border: none;
}

/* search button active state style */
.searchBarButton.btn-primary:not(:disabled):not(.disabled):active {
  background-color: var(--dark-orange);
}

.globalSearch {
  text-align: center;
}
