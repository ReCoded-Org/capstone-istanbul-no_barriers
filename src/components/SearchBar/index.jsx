import React from "react";
import { Button } from "react-bootstrap";
import "./index.css";
import { useTranslation } from "react-i18next";

const SearchBar = () => {
  const { t } = useTranslation();

  return (
    <div className="globalSearch">
      <h2>{t("resources.searchBar.searchTitle")}</h2>
      <div className="searchBar d-flex">
        <input
          className="searchBarInput flex-grow-1"
          type="text"
          placeholder={t("resources.searchBar.inputPlaceholder")}
          maxLength="15"
        />
        <Button className="searchBarButton shadow-none">
          {t("resources.searchBar.searchButton")}
        </Button>
      </div>
    </div>
  );
};

export default SearchBar;
