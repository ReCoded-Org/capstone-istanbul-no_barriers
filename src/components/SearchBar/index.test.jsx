import React from 'react';
import renderer from 'react-test-renderer';
import SearchBar from "./index";

it('SearchBar renders correctly', () => {
  const tree = renderer
    .create(<SearchBar />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});