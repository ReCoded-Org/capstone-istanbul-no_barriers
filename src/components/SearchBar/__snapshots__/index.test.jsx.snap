// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar renders correctly 1`] = `
<div
  className="globalSearch"
>
  <h2>
    resources.searchBar.searchTitle
  </h2>
  <div
    className="searchBar d-flex"
  >
    <input
      className="searchBarInput flex-grow-1"
      maxLength="15"
      placeholder="resources.searchBar.inputPlaceholder"
      type="text"
    />
    <button
      className="searchBarButton shadow-none btn btn-primary"
      disabled={false}
      type="button"
    >
      resources.searchBar.searchButton
    </button>
  </div>
</div>
`;
